xchange-ftx/src/main/java/org/knowm/xchange/ftx/service/FtxLendingServiceRaw.java

```diff
@@ -0,0 +1,277 @@
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount,
+          new FtxSubmitLendingOfferParams(coin, FtxAdapters.lendingRounding(new BigDecimal(sizeToLend)).doubleValue(), rate)
+      );
+      return new FtxLendDataDto(coin, info.getLocked(), info.getOffered(), sizeToLend, rate);
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't lend subaccount: " + subaccount + ", coin: " + coin + ", size: " + size + ", rate: " + rate, e);
+    }
+  }
+
+  public FtxLendDataDto lend(String subaccount, String coin, FtxLendingInfoFunction functionSize, FtxLendingInfoFunction functionRate) {
+    Objects.requireNonNull(coin);
+    if (StringUtils.isNotBlank(coin))
+      throw new FtxLendingServiceException("Coin are blank or empty");
+
+    FtxLendingInfoDto info = info(subaccount, coin);
+    if (info == null) {
+      throw new FtxLendingServiceException("No info for coin: " + coin);
+    }
+    double sizeToLend = functionSize.apply(info);
+    if (Double.compare(sizeToLend, info.getLendable()) == 1) {
+      throw new FtxLendingServiceException("Cant't lend > to lendable, subaccount: " + subaccount + ", coin: " + coin + ", sizeLendable: " + info.getLendable() + ", sizeToLend: " + sizeToLend);
+    }
+    if (sizeToLend < 0)
+      throw new FtxLendingServiceException("Cant't lend > to lendable, subaccount: " + subaccount + ", coin: " + coin + ", sizeLendable: " + info.getLendable() + ", sizeToLend: " + sizeToLend);
+
+    double rate = functionRate.apply(info);
+    if (rate < 0)
+      throw new FtxLendingServiceException("Cant't lend > to lendable, subaccount: " + subaccount + ", coin: " + coin + ", sizeLendable: " + info.getLendable() + ", sizeToLend: " + sizeToLend + ", rate: " + rate);
+
+    try {
+      ftx.submitLendingOffer(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount,
+          new FtxSubmitLendingOfferParams(coin, FtxAdapters.lendingRounding(new BigDecimal(sizeToLend)).doubleValue(), rate)
+      );
+      return new FtxLendDataDto(coin, info.getLocked(), info.getOffered(), sizeToLend, rate);
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't lend subaccount: " + subaccount + ", coin: " + coin + ", sizeToLend: " + sizeToLend + ", rate: " + rate, e);
+    }
+  }
+
+  public List<FtxLendingHistoryDto> histories(String subaccount) {
+    try {
+      return ftx.getlendingHistories(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount
+      ).getResult();
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't get lending infos subAccount: " + subaccount, e);
+    }
+  }
+
+  public List<FtxLendingHistoryDto> histories(String subaccount, List<String> coins) {
```

Comment:
similar to my comments above not addressed

Reply:
and below as well

Reply:
What do you mean not addressed?

Reply:
you need subAccount because your history are different for Each under account